packages/dds/sequence/src/revertibles.ts

```diff
@@ -114,13 +114,19 @@ export function appendDeleteIntervalToRevertibles(
 	string: SharedString,
 	interval: SequenceInterval,
 	revertibles: SharedStringRevertible[],
-) {
-	const startSeg = interval.start.getSegment() as SharedStringSegment;
+): SharedStringRevertible[] {
+	const startSeg = interval.start.getSegment() as SharedStringSegment | undefined;
+	if (!startSeg) {
```

Comment:
the observed behavior change here is that if an interval which becomes detached from the string is deleted, nothing is added to the revertible stack, right?

...and this should also not be relevant once intervals slide to an EndOfTree segment rather than detaching?

Reply:
I believe the answer is yes to both questions. I'm not sure why this didn't crash previously, though, since the type has always been `| undefined`.